packages/components/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,373 @@
+    colSpan?: HTMLTableCellElement['colSpan'];
+    /**
+     * Width of the column
+     */
+    width?: CSSProperties['width'];
+    /**
+     * Current sort direction of the column
+     */
+    sortDirection?: SortDirection;
+    /**
+     * Horizontal alignment of the content
+     * @default 'left'
+     */
+    align?: HorizontalAlignment;
+    /**
+     * Whether to truncate content with ellipsis when it overflows
+     * @default false
+     */
+    truncate?: boolean;
+    /**
+     * Aria label for asceding/descending sort. Variables: {column} - column name
+     * @default "Sort by {column} ascending/descending"
+     */
+    sortTranslations?: {
+        sortAscending?: string;
+        sortDescending?: string;
+    };
+    /**
+     * Whether the column should have a minimum width
+     * @default false
+     */
+    noShrink?: boolean;
+    /**
+     * Handler called when the sort direction changes
+     * @param direction - The new sort direction
+     */
+    onSortChange?: (direction: SortDirection) => void;
+    children: ReactNode;
+};
+
+export const TableHeaderCell = forwardRef<HTMLTableCellElement, TableHeaderCellProps>(
+    (
+        {
+            noShrink = false,
+            truncate = false,
+            align = 'left',
+            scope = 'col',
+            sortTranslations,
+            sortDirection,
+            colSpan,
+            width,
+            onSortChange,
+            children,
+        },
+        ref,
+    ) => {
+        const cellRef = useRef<HTMLTableCellElement>(null);
+        useSyncRefs<HTMLTableCellElement>(cellRef, ref);
+
+        useTextTruncation(cellRef);
+
+        const sortLabel = useMemo(() => {
+            if (typeof children === 'string') {
+                if (sortDirection === 'ascending') {
+                    return sortTranslations?.sortDescending ?? `Sort by ${children} descending`;
+                }
+                return sortTranslations?.sortAscending ?? `Sort by ${children} ascending`;
+            }
+
+            return sortDirection === 'ascending' ? 'Sort descending' : 'Sort ascending';
+        }, [children, sortDirection, sortTranslations]);
+
+        const handleSortChange = () => {
+            if (!onSortChange) {
+                return;
+            }
+
+            const newDirection: SortDirection =
+                sortDirection === undefined || sortDirection === 'descending' ? 'ascending' : 'descending';
+
+            onSortChange(newDirection);
+        };
+
+        return (
+            <th
+                ref={cellRef}
+                style={{ width }}
+                className={styles.headerCell}
+                scope={scope}
+                colSpan={colSpan}
+                data-align={align}
+                data-truncate={truncate}
+                data-no-shrink={noShrink}
+            >
+                {onSortChange ? (
+                    <button className={styles.sortButton} aria-label={sortLabel} onClick={handleSortChange}>
```

Comment:
please move the padding from the th to the button, else not the full cell is clickable